src/queries/pages.js

```diff
@@ -9,8 +9,8 @@ const AllPages = `
 `
 
 const SinglePage = `
-  query SinglePage($slug: String!) {
-    page(where: { slug: $slug }) {
```

Comment:
This works great, but there's another approach that I usually end up doing so that I don't have to update all my queries: create one token that only reads from draft and one token that only reads from published, and then, update the token in the fetch request based on it (usually it should be in just one place, not sure if it applies here, but something like this https://github.com/hygraph/docs-starter/blob/main/app/lib/hygraph.server.ts).

Reply:
Yeah, that's the better approach for sure. I'll go back and update. I'd originally (quick and dirty) used the gcms-stage header, but usually I know folks want draft content private, so the token works best all around

Reply:
Updates in [following commit](https://github.com/hygraph/hygraph-nextjs-blog-starter/pull/9/commits/1b0ede88cb8e3346b4d9ccba07353768917d7b60) to change to Token-based drafts